tests/test_data_source_connectors.py

```
@@ -0,0 +1,470 @@
+
+def test_builtin_connectors_register_without_explicit_import():
+    """Production wiring: a cold process that imports ONLY the registry (never the
+    connector module) must still resolve 'google_sheet'. Guards against the
+    self-register-on-import connector never being imported in prod.
+    """
+    code = (
+        "from app.services.data_sources import get_connector; "
+        "assert get_connector('google_sheet') is not None, 'not registered'; "
+        "print('REGISTERED')"
+    )
+    proc = subprocess.run(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)